Dockerfile

```diff
@@ -35,13 +35,17 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     xz-utils
 
 # Install Python deps
-RUN python3 -m pip install pipenv \
- && rm -rf ~/.cache
-
-COPY Pipfile Pipfile.lock /nextstrain/ncov-ingest/
-
-RUN PIPENV_PIPFILE=/nextstrain/ncov-ingest/Pipfile pipenv sync --system \
- && rm -rf ~/.cache
+RUN python3 -m pip install nextstrain-cli>=2.0.0 \
+    csv-diff>=0.6 \
+    fsspec \
+    pandas>=1.0.1 \
+    s3fs[boto3] \
+    aiobotocore[awscli,boto3] \
+    biopython \
+    regex \
+    envdir \
+    xopen \
+    typer
```

Comment:
At a minimum, let's at least put these into a file (e.g. the Python-conventional `requirements.txt`) so that they can be installed outside of the Docker image. The README doesn't explicitly discuss how to set up a non-containerized local dev env, but it does extensively discuss running the programs in such a local env. The use of Pipenv for that was implicit (though IIRC it used to be explicit?), implied by the presence of `Pipfile` and such. The presence of `requirements.txt` will be similar.

Ideally, it would still be nice to lock these deps so they're not changing and liable to break on new upstream releases updates. [I'd use](https://github.com/nextstrain/ncov-ingest/issues/413#issuecomment-1665064010) `pip-compile` from pip-tools for that. But I guess also ok not to bother with it for now.

Reply:
I switched to `requirements.txt` in [force-push](https://github.com/nextstrain/ncov-ingest/compare/04577fae6252ae48f845098c5f8a92e29d912657..2d94dc909a28cea157f2a166278028231c99a076). Started looking into `pip-compile` but didn't finish. This PR isn't blocking any other work at the moment, so I might leave it open and come back to this later.

Reply:
Merged with `requirements.txt` to unblock CI. Created an issue to use `pip-compile`: #438